control_unit.py

```diff
@@ -67,22 +67,33 @@ def _STR(instruction):
 
 
 def _RTI(instruction):
-    """unused"""
-    pass  # TODO throw exception here.
+    raise Exception("instruction is not implemented !")
+    pass  
 
 
 def _NOT(instruction):
-    """bitwise not"""
+    DR = (_NOT(instruction) >> 9) & 0X7                            # Destenation Register.
+    SR = (_NOT(instruction) >> 6) & 0X7                            # Source Register (register cntaining the data).
```

Comment:
same here, that's an endless recursive call.

Reply:
workin on it